src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/DictionaryDefaultConverter.cs

```diff
@@ -128,8 +128,9 @@ protected static JsonConverter<TValue> GetValueConverter(ref WriteStack state)
                         reader.ReadWithVerify();
 
                         // Get the value from the converter and add it.
-                        elementConverter.TryRead(ref reader, typeof(TValue), options, ref state, out TValue element);
-                        Add(element, options, ref state);
+                        bool success = elementConverter.TryRead(ref reader, typeof(TValue), options, ref state, out TValue element);
+                        Debug.Assert(success);
```

Comment:
I'm fine with the assert being added, but it doesn't actually help with any nullable validation, does it?

Reply:
I'm reverting all of the _System.Text.Json_ changes because Layomi's latest PR unifies the nullability attributes across their code base.